src/main/java/EPparser/CommandParser.java

```diff
@@ -0,0 +1,237 @@
+package EPparser;
+
+import commands.*;
+import Execution.CommandStack;
+import MovieUI.Controller;
+import MovieUI.MovieHandler;
+import wrapper.CommandPair;
+import commands.COMMANDKEYS;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+
+public class CommandParser {
+    /**
+     * Entry point to command parser Class
+     *
+     * @param command command that was entered by the user
+     */
+    public static void parseCommands(String command , Controller UIController) throws IOException {
+        command = command.toLowerCase();
+        String commandArr[] = command.split(" ");
+        rootCommand(commandArr , command, UIController);
+
+    }
+
+    /**
+     * Function to get the most probable root and sub root command if there was a typo in the user command
+     *
+     * @param CommandArr command that was entered by the user in split array form
+     * @param Command   command that was entered by the user.
+     * @param UIController the controller for the UI
+     */
+    public static void processCommand(CommandPair command , String[] CommandArr , String Command , Controller UIController) throws IOException {
```

Comment:
This is longer than 120 characters. Readers are forced to scroll horizontally --> hurts the readability.